### PR TITLE
Python docker caching

### PR DESCRIPTION
**Pull request by glogiotatidis**

### CODE REVIEW

**Comment by pmac** [comment]
Nice! Looks like good stuff. More cache plz 💯

File: .travis.yml [hunk 1]
```diff
@@ -3,13 +3,28 @@ language: python
 python: 2.7
 services:
   - docker
+before_install:
+  - echo "ENV GIT_SHA ${TRAVIS_COMMIT}" >> Dockerfile
```
Comment (safwanrahman) [inline .travis.yml]: Is it possible to do it with build arguments?

**Comment by safwanrahman** [comment]
So the approach can be something like building a docker image with git hash tag, and use the tag only to start the container with that hash.
I will try with another approach today.

**Comment by glogiotatidis** [comment]
Thanks!

**Pull request merged**